Entity/TransUnitRepository.php

```diff
@@ -210,6 +210,8 @@ protected function addTranslationFilter(QueryBuilder $builder, array $locales =
 
             if ((is_countable($ids) ? count($ids) : 0) > 0) {
                 $builder->andWhere($builder->expr()->in('tu.id', $ids));
+            } else {
+                $builder->andWhere($builder->expr()->eq(1, 0));
```

Comment:
Why?

Reply:
To avoid to return all results (because there is no where condition added) when there is no ids matching the search. In case there is no matching results I add a condition to the builder that is never true.

With the fix when the search doesn't matches, we have an empty list:
![image](https://github.com/lexik/LexikTranslationBundle/assets/652505/29dc9102-be16-40d7-b45e-a210abbfdfa7)

We didn't wanted to throw a NoResult exception as we are in ``addTranslationFilter`` and didn't wanted to change to much the logics but if you have a better idea, please suggest it.

Reply:
It should be in a separate PR if unrelated to the pure JS implementation

Reply:
@Seb33300 Right, I just revert 2 unrelated commit (this fix and the fix of deprecation)